### ISSUE TITLE
librpmbuild-sys backend?

### ISSUE THREAD

**Issue by tarcieri** [Richterrettich/rpm-rs#6]
Just discovered this project, which looks quite interesting! I just wanted to note that there's a `librpmbuild-sys` crate available:

- repo: https://github.com/rpm-software-management/librpm.rs/tree/master/librpmbuild-sys
- docs: https://rustrpm.org/librpmbuild_sys/index.html

There's presently no pure Rust idiomatic wrapper to this crate (or any plans to build one), but it seems like this crate could potentially provide such a wrapper (an optional feature) using its current API.

**Comment by Richterrettich** [comment]
Hi!
so you mean that rpm-rs should have an optional feature to use librpmbuild-sys as a backend?

**Comment by tarcieri** [comment]
Yeah, something like that, gated under a cargo feature. I'm not sure how that'd manifest at the API layer, but maybe you could make a trait for the actual RPM builder component and allow a `librpmbuild-sys` backend to be swapped in somehow.

Just a thought, mostly noting that this library solves the problem of what an idiomatic Rust API for an RPM builder should look like.

**Comment by Richterrettich** [comment]
I am open to the idea. If you want, you can sketch a PR. I think we need some sort of a proof of concept to see if this makes sense or not

**Comment by drahnr** [comment]
Could somebody kindly explain the gain of adding an impl of `librpmbuild-sys` to me, I seem to be oblivious to them.

**Comment by tarcieri** [comment]
@drahnr `librpmbuild` is effectively the reference implementation, so if nothing else, it'd be useful to compare whether the two backends produce equivalent outputs, but also it's more fully featured if people would like to leverage features that aren't yet implemented in `rpm-rs`

**Comment by drahnr** [comment]
@tarcieri but in that case one could also compare the output of both an compare the bytes / individual sections - without the need to introduce complexity to the library. On another note, I am not sure if it is a sane approach trying to implement _all_ features of rpm, there is a lot of legacy functionality which is effectively dead.
